### DISCUSSION TITLE
mutateRecordDataUsing seems not working for a Replicate action?

### DISCUSSION THREAD

**Discussion by gilr** [filamentphp/filament#12617] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.73

### How can we help you?

In a view page, the 'excludeAttributes' removes the name, but the 'mutateRecordDataUsing' does not change anything. Did I make an error? I found a solution, using 'BeforeReplicaSaved', and it works (see the code in the bottom), but I would like to understand.

```
   protected function getHeaderActions(): array
    {
        return [
            Action::make('back')
                ->label ('Back to the list')
                ->url(static::getResource()::getUrl())
                ->button()
                ->color('info'),
            Actions\EditAction::make(),
            Actions\ReplicateAction::make()
                ->excludeAttributes(['name'])
                ->mutateRecordDataUsing(function (array $data): array {
                    $data['name'] = "Clone";
                    $data['created_by'] = Auth()->id();
                    $data['updated_by'] = Auth()->id();
                    $data['created_at'] = now();
                    $data['updated_at'] = now();
                    return $data;
                })
                ->successRedirectUrl(fn (Model $replica): string => route('filament.admin.resources.publications.edit', [
                    'record' => $replica,
                ]))
        ];
    }

```

But is works perfectly using a beforeReplacaSaved instead of mutateRecordDataUsing:

```
   protected function getHeaderActions(): array
    {
        return [
            Action::make('back')
                ->label ('Return to the list')
                ->url(static::getResource()::getUrl())
                ->button()
                ->color('info'),
            Actions\EditAction::make(),
            Actions\ReplicateAction::make()
                ->excludeAttributes(['name'])
                ->beforeReplicaSaved(function (Model $replica, Model $record): void {
                    $replica->name = "CLONE:" . $record->name;
                    $replica->created_by = Auth()->id();
                    $replica->updated_by = Auth()->id();
                    $replica->created_at = now();
                    $replica->updated_at = now();
                })
                ->successRedirectUrl(fn (Model $replica): string => route('filament.admin.resources.publications.edit', [
                    'record' => $replica,
                ]))
        ];
    }

```
Any ideas?

**Comment by stevecrossuk** [comment]
I have the same issue, and had to use "beforeReplicaSaved" in order to achieve the same functionality.

`ReplicateAction::make()
                    ->beforeReplicaSaved(function (Model $replica, Model $record): void {
                        $replica->name = "CLONE:" . $record->name;
                    }),`

**Comment by intrepidws** [comment]
I ran into the same problem, thanks for the suggestion @stevecrossuk

**Comment by Boefjim** [comment]
Just pitching in that I am also having the same problem (with package version v3.2.92).
Using `beforeReplicaSaved` seems to still be a functioning workaround for now.

**Reply by Boefjim** [reply]
I have created an issue for it, so hopefully it gets fixed soon: https://github.com/filamentphp/filament/issues/13604

**Reply by intrepidws** [reply]
Thanks for creating the issue @Boefjim

**Comment by Boefjim** [comment, marked as answer]
For anyone coming across this in the future, @intrepidws figured out why `mutateRecordDataUsing` is not functioning as you might expect here: https://github.com/filamentphp/filament/issues/13604#issuecomment-2259131733

**Comment by photojo** [comment]
Version 3.2.135 
I'm working since several hours on following problem: 
`      ReplicateAction::make('duplicate')
                    ->form([
                        TextInput::make('first')->dehydrated(false)
                    ])
                    ->beforeReplicaSaved(function (Customer $replica, array $data,): void {
                        $replica->fill([
                            'name' => $data['first']
                        ]);
                    }),`
                    
                    
Error: Undefined array key "first"

But with 
`...
TextInput::make('first')->dehydrated(true)
...`

I get Error:

Unknown column 'first' in 'field list' (Connection: mysql, SQL: insert into `customers`
because 'first' is not in DB-table.


I never had any problem in Filament 2.

Any idea?!
